API.md

```diff
@@ -77,32 +77,74 @@ What user date is provided by default via Auth0?
 
 ## Elections API
 
-### `GET /elections/list`
+### `POST /election/`
+
+```
+{
+  name: string
+}
+```
+
+returns
+
+```
+{
+  id: string // the newly minted election UUID (not meaningful to the user, should not be displayed)
```

Comment:
```suggestion
  id: string, // the newly minted election UUID (not meaningful to the user, should not be displayed)
  name: string,
  createdAt: datetime,
```

Reply:
On second thought, to keep things standardized, why wouldn't this return the same as `GET /elections/<election_id>` endpoint?

Reply:
we can, most of the time I think returning just the ID is what you want at this stage, and then query `/elections/<election_id>`. My approach to APIs is to keep them minimal and reduce the number of different paths one might choose, because it's hard to remove support for alternatives paths through the API.

Reply:
Ok, I understand. I guess I'm looking at this from a user perspective as to which of the two scenarios is more "minimal":

1. post to `/elections/` to get id
2. get `/elections/:id` to get election record

vs

1. post to `/elections/` to get election record.

Reply:
Right, but I think how it will happen is more:

1. post to `/elections/` to create an election, get the ID
2. redirect (client-side) to `/elections/<election_id>`
3. now either you make the network request or you stashed the info from the previous API call and you render it either way (<-- that's two different ways of using the API.)

Reply:
Or… per spec update PR…

1. post to /elections/ to create an election, get the ID
2. force re-render /elections/ screen which would GET the list again.

I'll follow you lead here! 😄

Reply:
we could do that, sure, but I think my point would remain: wouldn't you just re-query `/elections/` to list all of them? I don't think you'd try to combine the old list with the new record all on the client, right?

Reply:
I'm convinced. Thanks for taking the time to explain.